### PR TITLE
Add Source instances

### PR DESCRIPTION
**Pull request by bregman-arie**
Before this change, System.sources is set to a dictionary
of sources as loaded from the configuration file.

This change adds instead Source() instances to System.sources.

### CODE REVIEW

File: cibyl/models/ci/system.py [hunk 1]
```diff
@@ -79,6 +79,14 @@ def add_job(self, job: Job):
         """
         self.jobs.append(job)
 
+    def add_source(self, source: Source):
+        """Add a source to the CI system
+
+        :param source: Source to add to the system
+        :type job: Source
```
Comment (cescgina) [inline cibyl/models/ci/system.py:86]: s/job/:class:`.Source` this adds a reference to the class in sphinx

File: cibyl/sources/jenkins_job_builder.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def __init__(self, url: str, dest: str = None, branch: str = None):
         :param branch: Branch to checkout
         :type branch: str
         """
-        super().__init__("", url)
+        super().__init__("", url=url, driver="jenkins_job_builder")
```
Comment (cescgina) [inline cibyl/sources/jenkins_job_builder.py:50]: We should add the driver attribute to the rest of sources, but we can do that in another PR

**Review by cescgina — CHANGES_REQUESTED** [review]
Looks good, a couple of minor comments

File: cibyl/sources/jenkins_job_builder.py (hunk 2 above)
Comment (bregman-arie) [reply cibyl/sources/jenkins_job_builder.py:50]: Yes, but right now only jjb and jenkins I think inherit from Source

Comment (cescgina) [reply cibyl/sources/jenkins_job_builder.py:50]: Yes, I think that's true, we can leave it for a  later PR

**Review by cescgina — APPROVED** [review]
All good!
